File: mobile/lib/models/piece.dart
class Piece {
  final String id;
  final String displayNumber;
  final int seriesValue;
  final int referencePinceauxValue;
  final String colorPrimary;
  final String? colorSecondary;
  final Map<String, dynamic>? colorSignature;
  final String? materialNotes;
  final DateTime creationDate;
  final String? artistNote;
  final String? currentOwnerId;
  final String status;
  final String? photoUrl;

  Piece({
    required this.id,
    required this.displayNumber,
    required this.seriesValue,
    required this.referencePinceauxValue,
    required this.colorPrimary,
    this.colorSecondary,
    this.colorSignature,
    this.materialNotes,
    DateTime? creationDate,
    this.artistNote,
    this.currentOwnerId,
    this.status = 'non_distribue',
    this.photoUrl,
  }) : creationDate = creationDate ?? DateTime.now();

  factory Piece.fromJson(Map<String, dynamic> json) {
    return Piece(
      id: json['id'] as String,
      displayNumber: json['display_number'] as String,
      seriesValue: json['series_value'] as int,
      referencePinceauxValue: json['reference_pinceaux_value'] as int,
      colorPrimary: json['color_primary'] as String,
      colorSecondary: json['color_secondary'] as String?,
      colorSignature: json['color_signature'] as Map<String, dynamic>?,
      materialNotes: json['material_notes'] as String?,
      creationDate: json['creation_date'] != null
          ? DateTime.parse(json['creation_date'] as String)
          : DateTime.now(),
      artistNote: json['artist_note'] as String?,
      currentOwnerId: json['current_owner_id'] as String?,
      status: (json['status'] as String?) ?? 'non_distribue',
      photoUrl: json['photo_url'] as String?,
    );
  }
}

File: app/models/piece.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime, Enum, Text, ForeignKey, JSON
from sqlalchemy.orm import relationship
from ..database import Base
import enum


class PieceStatus(str, enum.Enum):
    non_distribue = "non_distribue"
    active = "active"
    lost = "lost"


class ColorEnum(str, enum.Enum):
    blanc = "blanc"
    gris = "gris"
    noir = "noir"
    jaune = "jaune"
    bleu = "bleu"
    vert = "vert"
    rouge = "rouge"
    magenta = "magenta"
    multicolore = "multicolore"


class Piece(Base):
    __tablename__ = "pieces"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    display_number = Column(String(10), unique=True, nullable=False)
    series_value = Column(Integer, nullable=False)
    reference_pinceaux_value = Column(Integer, nullable=False)
    color_primary = Column(Enum(ColorEnum), nullable=False)
    color_secondary = Column(Enum(ColorEnum), nullable=True)
    color_signature = Column(JSON, nullable=True)
    color_signature_history = Column(JSON, default=list)
    texture_signature = Column(JSON, nullable=True)
    top_image = Column(Text, nullable=True)
    material_notes = Column(Text, nullable=True)
    creation_date = Column(DateTime, default=datetime.utcnow)
    artist_note = Column(Text, nullable=True)
    current_owner_id = Column(String(36), ForeignKey("users.id"), nullable=True)
    status = Column(Enum(PieceStatus), default=PieceStatus.non_distribue)
    photo_url = Column(String(500), nullable=True)

    current_owner = relationship("User", back_populates="owned_pieces")
    scan_events = relationship("ScanEvent", back_populates="piece")
    provenance_events = relationship("ProvenanceEvent", back_populates="piece")
